Analysis/output/model_failures_report.Rmd

```diff
@@ -0,0 +1,177 @@
+
+cases_check <- list()
+rhat_check <- list()
+
+for (i in 1:length(all_configs)){
+  config_filename <- all_configs[i]
+  config <- yaml::read_yaml(config_filename)
+  file_names <- taxdat::get_filenames(config, params$cholera_directory)
+  cntry_code <- stringr::str_remove(unlist(strsplit(config_filename, "/"))[length(unlist(strsplit(config_filename, "/")))], ".yml")
+  print(paste("Reading files associated with", config_filename))
+
+  ## get cases
+  try({
+    model.rand <- taxdat::read_file_of_type(file_names[["stan_output"]], "model.rand")
+    niter_per_chain <- dim(MCMCvis::MCMCchains(model.rand, params='lp__', chain_num=1))[1]
+    nchain <- dim(MCMCvis::MCMCchains(model.rand, params='lp__'))[1] / niter_per_chain
+    stan_output <- lapply(rstan::extract(model.rand), function(x){array(x,c(niter_per_chain, nchain, dim(x)[-1]))})
+
+    cases_chains <- apply(stan_output$grid_cases, c(3,2), mean)
+    # rates_chains <- apply(exp(stan_output$log_lambda), c(3,2), mean)
+    # pop_est <- cases_chains[,1]/rates_chains[,1]
+    total_cases_chains <- apply(cases_chains, 2, sum)
+
+  })
+
+  try({
+    covar_cube_output <- taxdat::read_file_of_type(file_names[["covar"]], "covar_cube_output")
+    pop_layer <- covar_cube_output$covar_cube[,,1]
+
+    cases_summ <- data.frame(cntry = cntry_code, config = config_filename, pop = sum(pop_layer[,1]), chain = 1:length(total_cases_chains), cases_mean = total_cases_chains) 
+  })
+
+  try({
+    fit_summary <- rstan::summary(model.rand)
+    rhats <- tibble::tibble(Rhat = round(fit_summary$summary[which(stringr::str_detect(row.names(fit_summary$summary), 
+        "modeled_cases")), "Rhat"], 2)) %>% dplyr::mutate(x = dplyr::row_number())
+    rhat_thresh <- 1.05
+    frac_above <- sum(rhats$Rhat > rhat_thresh)/nrow(rhats)
+
+    rhat_summ <- data.frame(cntry = cntry_code, config = config_filename, frac_above = frac_above)
+  })
+
+  try({
+    if("list" %in% class(cases_check)){
+      cases_check <- cases_summ
+    } else {
+      cases_check <- rbind(cases_check, cases_summ)
+    }
+  })
+
+  try({
+    if("list" %in% class(rhat_check)){
+      rhat_check <- rhat_summ
+    } else {
+      rhat_check <- rbind(rhat_check, rhat_summ)
+    }
+  })
+
+}
+
+```
+
+# Check if cases exceed population
+
+```{r cases_check}
+
+pop_failures <- dplyr::mutate(cases_check, 
+  failure_pop = ifelse(cases_mean > pop, TRUE, FALSE)) %>%
+  dplyr::mutate(cp_label = paste(round(cases_mean,0), sep = "/"))
+
+```
+
+```{r plot_pop_failures, fig.cap = "Failures for population check (cases/population)"}
+ggplot2::ggplot(pop_failures, ggplot2::aes(x = chain, y = cntry)) +
+  ggplot2::geom_tile(ggplot2::aes(fill = failure_pop)) +
+  ggplot2::geom_label(ggplot2::aes(label = cp_label))
+
+```
+
+```{r plot_pop, fig.cap = "Population estimates"}
+pop_df <- dplyr::distinct(pop_failures, cntry, pop) %>%
+  dplyr::mutate(x_tick = 3E8)
+
+ggplot2::ggplot(pop_failures, ggplot2::aes(y = cntry)) +
```

Comment:
Is this a place we could use the taxdat scale functions? (Not saying it is, just not sure)

Reply:
I'm not sure what taxdat scale functions you're referring to -- the map scales? this is just a population figure.